parlai/core/params.py

```diff
@@ -362,17 +376,36 @@ def add_parlai_args(self, args=None):
             help='Whether the agent should preprocess the data while building'
                  'the pytorch data')
         pytorch.add_argument(
-            '--batch-sort-cache', type=str,
-            choices=['pop', 'index', 'none'], default='none',
-            help='Whether to have batches of similarly sized episodes, and how'
-            'to build up the cache')
+            '-pybsrt', '--pytorch-teacher-batch-sort',
+            type='bool', default=False,
+            help='Whether to construct batches of similarly sized episodes'
+            'when using the PytorchDataTeacher (either via specifying `-pyt`'
+            'or `-pytd`')
+        pytorch.add_argument(
+            '--batch-sort-cache-type', type=str,
+            choices=['pop', 'index', 'none'], default='pop',
+            help='how to build up the batch cache')
         pytorch.add_argument(
             '--batch-length-range', type=int, default=5,
             help='degree of variation of size allowed in batch')
         pytorch.add_argument(
             '--shuffle', type='bool', default=False,
             help='Whether to shuffle the data')
-        self.add_parlai_data_path(parlai)
+        pytorch.add_argument(
+            '--batch-sort-field', type=str, default='text',
+            help='What field to use when determining the length of an episode')
+        pytorch.add_argument('-pyclen', '--pytorch-context-length', default=-1,
+                             type=int,
+                             help='Number of past utterances to remember when '
+                                  'building flattened batches of data in multi-'
+                                  'example episodes.'
+                                  '(For use with PytorchDataTeacher)')
+        pytorch.add_argument('-pyincl', '--pytorch-include-labels',
+                             default=True, type='bool',
+                             help='Specifies whether or not to include labels '
+                                  'as past utterances when building flattened '
+                                  'batches of data in multi-example episodes.'
+                                  '(For use with PytorchDataTeacher)')
```

Comment:
should we really rename these or just take over the existing args in the pytorch teacher?

Reply:
I guess that makes it hard to phase out the other one rather than just replacing it